### PR TITLE
[Ops][Feature] Refactor Ascend Sampler to use torch_npu top-k top-p sampling

### PR DESCRIPTION
**Pull request by kichanll**
### What this PR does / why we need it?
This PR refactors the `AscendTopKTopPSampler` to use the native `torch_npu.npu_top_k_top_p_sample` operator instead of custom C++ ops. It introduces a helper function `generate_random_sequence` to asynchronously generate exponential random variables for sampling. Additionally, the PR updates `_apply_top_k_top_p_ascendc` to leverage the new NPU sampling operator.

Feedback:
- In `generate_random_sequence`, use `torch.npu.current_stream()` instead of `torch.npu.default_stream()` to wait for the stream, preventing potential race conditions in multi-stream environments.
- In `_apply_top_k_top_p_ascendc`, ensure that `p` and `k` are explicitly moved to the same device as `logits` to avoid device mismatch runtime errors.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
New unit tests have been added in `tests/ut/sample/test_sampler.py` covering various sampling configurations and helper functions.

- vLLM version: v0.21.0
- vLLM main: https://github.com/vllm-project/vllm/commit/9090368b650896bf5fc990c921df7eb4c20355a5

### CODE REVIEW

File: vllm_ascend/sample/sampler.py [hunk 1]
```diff
@@ -132,36 +149,31 @@ def forward_native(self, logits, generators, k, p):
         """Override pytorch native implementation to torch_npu"""
         # when batch_invariant mode is enabled, we should use vllm's implementation.
         # or it will make batch_invariant mode not working.
-        if envs.VLLM_BATCH_INVARIANT:
-            return super().forward_native(logits, generators, k, p)
```
Comment (Ronald1995) [inline vllm_ascend/sample/sampler.py:136]: why delete this？this branch is to ensure batch invariance of inferencing.

File: vllm_ascend/sample/sampler.py [hunk 2]
```diff
@@ -15,6 +15,22 @@
 _SAMPLING_EPS = 1e-5
 
 
+def generate_random_sequence(
+    logits: torch.Tensor,
+    generators: dict[int, torch.Generator],
+    stream: torch.npu.Stream,
+) -> torch.Tensor:
```
Comment (Ronald1995) [inline vllm_ascend/sample/sampler.py]: please add docstring for this public func

File: vllm_ascend/sample/sampler.py (hunk 1 above)
Comment (kichanll) [reply vllm_ascend/sample/sampler.py:136]: This branch code has been restored.

File: vllm_ascend/sample/sampler.py (hunk 2 above)
Comment (kichanll) [reply vllm_ascend/sample/sampler.py]: done
